src/graphic/text/rt_render.cc

```diff
@@ -307,28 +323,24 @@ class RenderNode {
  */
 class Layout {
 public:
-	explicit Layout(std::vector<std::shared_ptr<RenderNode>>& all)
-	   : h_(0), idx_(0), all_nodes_(all) {
+	explicit Layout(std::vector<RenderNode*>& all) : h_(0), idx_(0), all_nodes_(all) {
```

Comment:
This change from `std::shared_ptr` to plain pointer is beyond my knowledge of C++. Looks consistent though.